interfaces/vault_kv/v0/README.md

```diff
@@ -24,7 +24,7 @@ Provider expectations
 - Is expected to provide a ca certificate used to validate the vault server's certificate.
 - Is expected to provide a key value mount, the mount name shall respect the following pattern: `charm-<requirer app>-<requirer provided suffix>`
 - Is expected to create an approle restricted to the requiring unit's egress subnet.
-- Is expected to create a Juju secret containing a role-id and role-secret-id for each unit.
+- Is expected to create a Juju secret containing a role-id and role-secret-id for each unit. If `wrap_ttl` is requested by the Requirer, it is expected to return a response-wrapping token instead of the `role-secret-id`.
```

Comment:
This probably seems trivial to infer, but I think we need to be clear here about what the contents of the response-wrapping token are.

I'm assuming you intend it to be the response of [the call which creates the secret id](https://developer.hashicorp.com/vault/api-docs/auth/approle#generate-new-secret-id)? As in, the contents will look like this (in the successful case)?

```
{
  "auth": null,
  "warnings": null,
  "wrap_info": null,
  "data": {
    "secret_id_accessor": "84896a0c-1347-aa90-a4f6-aca8b7558780",
    "secret_id": "841771dc-11c9-bbc7-bcac-6a3945a69cd9",
    "secret_id_ttl": 600,
    "secret_id_num_uses": 50
  },
  "lease_duration": 0,
  "renewable": false,
  "lease_id": ""
}
```

Reply:
What about providing a link to the official Vault documentation? Being more specific, the approle usage best practices has a dedicated section for it: https://developer.hashicorp.com/vault/tutorials/auth-methods/approle-best-practices?in=vault%2Fauth-methods#approle-response-wrapping

Reply:
@skatsaounis Sorry, to be clear I don't think we should actually enumerate the fields. Just that we should specifically say which API response this is.

Reply:
please take a look at the updated version of the PR and let me know :slightly_smiling_face:

Reply:
I apologise as I was unclear in my recommendation. I should have been more explicit. I suggest that we mention the API call directly, something along the lines of this.

```
If `wrap_ttl` is requested by the Requirer, the Provider is expected to return a response-wrapping token instead of the `role-secret-id`. The response-wrapping token must contain the response of the `POST	/auth/approle/role/:role_name/secret-id` API call.
```

The reason I feel it is important to mention this is because a response-wrapping token may contain _any_ Vault API response, and we want to be explicit about what the Requirer can expect here. Again, I know the reader / implementer can likely infer this, but I'd like to remove the chance for a misinterpretation here.

Reply:
Hopefully the last version can resolve this thread. Please take a look :slightly_smiling_face:

Reply:
I think that helps, thanks.

Seems I can't resolve the thread, I guess due to permissions.